Order view - 배송지 정보 추가/삭제/보기 API

joey-younggap opened the pull request:
[Add] 배송지 정보 추가/삭제/보기 API (테스트 완료)
- 해당 View 작성 및 order urls.py에 url 추가함

[Fix] Product view 수정
- ProductView Get 함수에서 기존 option_name만 리스트로 전달하던 것을 option_id, option_name으로
딕셔너리로 전달 (추후 장바구니에서 프론트엔드에서 option_id 값 전달이 필요하기 때문)

Commit:
[Add] Order View - 배송지 API 추가
- 배송지 정보 추가/삭제/보기 API 작성(테스트 완료)
- Order urls.py에 배송지 위한 url 추가

[Fix] Product view 수정
- option 값 전달 시 리스트로 option_name만 주던 것을 딕셔너리로 option_id, option_name 전달
  (추후 장바구니에서 cart_item 정보를 불러올 때option_id 값이 필요하기 때문임)

soheon-lee commented on order/views.py:
```diff
@@ -2,23 +2,26 @@
 
-            cart = Cart.objects.filter(customer_id=request.customer.id).last()
-            cart_items = CartItem.objects.filter(cart_id=cart.id, prodcut_id=data['product_id'])
+            if Cart.objects.filter(customer_id=data['customer_id']).exists():       # 해당 ID의 카트가 존재하면, 그거 가져옴
+                cart = Cart.objects.filter(customer_id=data['customer_id']).last()
```
filter().last() 사용하지 않고 다른 메소드 get() 찾아주세요

joey-younggap replied:
수정 사항 반영해서 다시 푸쉬 올렸습니다. :)

soheon-lee commented on order/views.py:
```diff
@@ -42,28 +45,67 @@ def post(self, request):
     def get(self, request):
-        cart = Cart.objects.filter(customer_id=request.customer.id).last()
-        cart_items = CartItem.objects.get(cart_id=cart.id).cartitem_set.all()
+        cart = Cart.objects.filter(customer_id=request.GET.get('customer_id')).last()
```
마찬가지 filter().last()

joey-younggap replied:
수정사항 반영해서 푸쉬 드렸습니다.

Commit:
wip

Commit:
[Fix] Order View 중 코멘트 사항 수정
- filter().last() get으로 가져오도록 로직 수정

Commit:
오타수정

soheon-lee commented on order/views.py:
```diff
@@ -42,28 +45,67 @@ def post(self, request):
-
-        CartItems.objects.filter(cart_id=cart.id, product_id=data['product_id']).delete()
+        
+        cart = Cart.objects.filter(customer_id=data['customer_id']).last()
```
.filter().last()

Commit:
wip

Commit:
[Add] Order view 추가
- Order 생성 후 저장
- 기존 Cart는 상태가 주문완료로 변경되며, 더 이상 장바구니에 노출되지 않도록 함

[Fix] Order 모델 수정
- Cart에 cart_status 추가

soheon-lee commented on order/models.py:
```diff
@@ -31,6 +31,7 @@ class Meta:
 
 class Cart(models.Model):
     customer   = models.ForeignKey('customer.Customer',on_delete=models.SET_NULL,null=True)
+    cart_status = models.ForeignKey('OrderStatus', on_delete=models.SET_NULL, null=True)
     created_at = models.DateTimeField(auto_now_add=True)
```
코드 간격 맞춰주세요

soheon-lee commented on order/views.py:
```diff
@@ -38,35 +39,165 @@ def post(self, request):
+
+            cart_items = [
+                {
+                    'cart_item_id' : cart_item.id,
+                    'product_id' : cart_item.product.id,
+                    'name' : cart_item.product.name,
+                    'sub_img_url' : cart_item.product.sub_img_url,
+                    'selected_option' : OptionItems.objects.get(id=cart_item.selected_option_id).name,
+                    'sales_price' : cart_item.product.sales_price,
+                    'per_quantity' : cart_item.product.sales_price_comment.split('(')[1].replace(')', ' 기준'),
+                    # sales_price_comment 기준가 xx원 (600g) 에서 600g 파싱한 뒤,' 기준' 멘트 추가 
+                    'quantity' : cart_item.quantity
```
코드 간격 맞춰주세요

soheon-lee commented on order/views.py:
```diff
@@ -38,35 +39,165 @@ def post(self, request):
+            ]
+
+            cart = [
+                {
+                    'cart_id' : cart.id,
+                    'total_amount' : total_amount,
+                    'delivery_cost' : delivery_cost,
+                    'expected_amount' : expected_amount
```
코드 간격 맞춰주세요

soheon-lee commented on order/views.py:
```diff
@@ -38,35 +39,165 @@ def post(self, request):
-                'sales_price' : cart_item.product.sales_price,
-                'quantity' : cart_item.quantity
-            } for cart_item in cart_items
+                'address_id' : delivery_address.id,
+                'name' : delivery_address.name,
+                'phone' : delivery_address.phone,
+                'address' : delivery_address.address
+            } for delivery_address in delivery_addresses
```
코드 간격 맞춰주세요

soheon-lee commented on order/views.py:
```diff
@@ -38,35 +39,165 @@ def post(self, request):
+    @login_decorator
+    def post(self, request):
+        try:
+            data = json.loads(request.body)
+            customer_id = request.customer.id
+
+            cart = Cart.objects.get(customer_id=customer_id)
+            delivery_address = DeliveryAddress.objects.get(customer_id=customer_id)
+            expected_amount = Decimal(data['expected_amount'])
```
코드 간격 맞춰주세요

Commit:
코드 간격 맞추기

soheon-lee commented:
finally LGTM
